containers/test-apps/courier/src/main/java/com/twosigma/waiter/courier/GrpcClient.java

```diff
@@ -0,0 +1,280 @@
+            public <ReqT, RespT> ClientCall<ReqT, RespT> interceptCall(final MethodDescriptor<ReqT, RespT> method,
+                                                                       final CallOptions callOptions,
+                                                                       final Channel next) {
+                return new SimpleForwardingClientCall<ReqT, RespT>(next.newCall(method, callOptions)) {
+
+                    @Override
+                    public void start(final Listener<RespT> responseListener,
+                                      final Metadata headers) {
+                        super.start(new SimpleForwardingClientCallListener<RespT>(responseListener) {
+                            @Override
+                            public void onHeaders(final Metadata headers) {
+                                logFunction.apply("headers received from server:" + headers);
+                                super.onHeaders(headers);
+                            }
+
+                            @Override
+                            public void onClose(final Status status, final Metadata trailers) {
+                                logFunction.apply("status received from server:" + status);
+                                logFunction.apply("trailers received from server:" + trailers);
+                                super.onClose(status, trailers);
+                            }
+                        }, headers);
+                    }
+                };
+            }
+        });
+    }
+
+    public static CourierReply sendPackage(final String host,
+                                           final int port,
+                                           final Map<String, Object> headers,
+                                           final String id,
+                                           final String from,
+                                           final String message) throws InterruptedException {
+        final ManagedChannel channel = initializeChannel(host, port);
+
+        try {
+            final Channel wrappedChannel = wrapResponseLogger(channel);
+            final Metadata headerMetadata = attachRequestHeaders(headers);
+
+            final CourierGrpc.CourierFutureStub rawStub = CourierGrpc.newFutureStub(wrappedChannel);
+            final CourierGrpc.CourierFutureStub futureStub = MetadataUtils.attachHeaders(rawStub, headerMetadata);
+
+            logFunction.apply("will try to send package from " + from + " ...");
+            final CourierRequest request = CourierRequest
+                .newBuilder()
+                .setId(id)
+                .setFrom(from)
+                .setMessage(message)
+                .build();
+            final CourierReply response;
+            try {
+                response = futureStub.sendPackage(request).get();
+            } catch (final StatusRuntimeException e) {
+                logFunction.apply("RPC failed, status: " + e.getStatus());
```

Comment:
Should we be making direct calls to a `LOGGER` in nearly every case that `logFunction` is currently being used?